packages/zoe/src/contractFacet/exit.js

```diff
@@ -2,65 +2,116 @@
 
 import { assert, details as X, q } from '@agoric/assert';
 import { E } from '@endo/eventual-send';
-import { Far } from '@endo/marshal';
+import { vivifyFarClass, provideDurableSetStore } from '@agoric/vat-data';
+import { M, initEmpty } from '@agoric/store';
 
 import {
   isOnDemandExitRule,
   isAfterDeadlineExitRule,
   isWaivedExitRule,
 } from '../typeGuards.js';
 
-/**
- * Makes the appropriate exitObj, which runs in ZCF and allows the seat's owner
- * to request the position be exited.
- */
+const ExitObjectGuard = M.interface('ExitObject', { exit: M.call().returns() });
```

Comment:
What differentiates naming like `<Noun>Guard = M.interface(…)` vs. `<Noun>I = M.interface(…)` vs. `<Noun>Shape = M.interface(…)`? This seems related to https://github.com/Agoric/agoric-sdk/pull/6502#discussion_r1014543291 , and I'd love to have some robust conventions.

Reply:
In some other comment thread, I made a few suggestions and IIRC @Chris-Hibbert 's favorite was also mine. So absent further dissent, I'll proceed assuming agreement.

`<Noun>I = M.interface(...)`. In case the font isn't clear, that's a capital i suffix, short for Interface. We both like this better than `<Noun>Guard`, `<Noun>Interface`, or `<Noun>InterfaceGuard`, even though all of these are not incorrect. "Guard" by itself is less specific, as we also have method guards and await guards, although we rarely if even need to name these. Thus, this example should be

```suggestion
const ExitObjectI = M.interface('ExitObject', { exit: M.call().returns() });
```

`<Noun>Shape` should only be used for patterns, not guards. So somewhere we may have a related
```js
const ExitObjectShape = M.remotable('ExitObject');
```
Interfaces are implemented by classes or instances.
InterfaceGuards consist of named MethodGuards.
Aside from the special `M.await(pattern)` await guard, methodGuards use patterns to check their arguments and results.
Values are tested against patterns/shapes, and either pass or fail.

Reply:
Started documentation at https://github.com/Agoric/agoric-sdk/wiki/Interfaces , please feel free to elaborate and/or correct my mistakes.

Reply:
I added an example of an interface for a faceted class.